### PR TITLE
Moving from throwing model to returning nil

### PR DESCRIPTION
**Pull request by mergesort**
This PR changes `public func get()` from being a throwing function to a function that returns `T?` instead.

Doing a do/try catch around everything that can live in a cache doesn't feel semantically correct. In a cache, the absence of a value is not representative of an error, but rather is a condition that feels like quite ordinary.

On the other hand, instead of silencing errors when calling `public func clear()`, instead it surfaces that error and now requires a `do/catch`, so that way you can see if something went wrong while clearing the cache.

Don't hesitate to let me know if there's anything I missed, or can be improved upon.
Thanks!

### CODE REVIEW

**Comment by asowers1** [comment]
Hi Joe, thanks for taking the time to go over everything -- looks great. Would you mind updating the README to reflect these changes? :)

**Comment by mergesort** [comment]
Of course not. Pushed.

File: README.md [hunk 1]
```diff
@@ -109,11 +109,8 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationC
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
 
-        do {
-            let settings = try appSettings.get()
+        if let settings = try appSettings.get() {
```
Comment (asowers1) [inline README.md]: Wouldn't this `try` in get be extraneous? There's a few other `try`s on `gets` in here.

**Comment by asowers1** [comment]
Also, before it's merged, you're welcome to add yourself in the `Authors and Contributors` section.

**Comment by mergesort** [comment]
Aw, thank you! Added myself.

**Comment by asowers1** [comment]
👍

**Comment by asowers1** [comment]
Cocoapods pointing to https://github.com/asowers1/CodableCache/releases/tag/0.2.0

**Pull request merged**